### PR TITLE
feat: Initial implementation of the charm

### PR DESCRIPTION
**Pull request by ghislainbourgeois**
# Description

Initial implementation of the charm. This does not currently implement the future `fiveg_rfsim` relation and instead hard codes the DU address. Also, it does not implement any actions to run tests, those will be added in future PRs.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+            logger.info("Waiting for Pod IP address to be available")
+            return
+        self.unit.set_workload_version(self._get_workload_version())
+        if not self._container.exists(path=BASE_CONFIG_PATH):
+            event.add_status(WaitingStatus("Waiting for storage to be attached"))
+            logger.info("Waiting for storage to be attached")
+            return
+        event.add_status(ActiveStatus())
+
+    def _configure(self, _) -> None:  # noqa C901
```
Comment (gruyaume) [inline src/charm.py:92]:
```suggestion
    def _configure(self, _) -> None:
```

File: tests/unit/fixtures.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from unittest.mock import patch
+
+import pytest
+import scenario  # type: ignore[import-untyped]
```
Comment (gruyaume) [inline tests/unit/fixtures.py:8]: Is the ignore needed?

**Review by gruyaume — APPROVED** [review]
Overall this looks good, just a couple of nits

File: tests/unit/fixtures.py (hunk 2 above)
Comment (ghislainbourgeois) [reply tests/unit/fixtures.py]: No, removed

**Pull request merged**